base-template.sh

```diff
@@ -15,7 +15,8 @@
 #   0 when successfully finished, 1 otherwise
 #####################################################
 process_git_hook() {
-    set_main_variables
+    set_main_variables "$@" || return 1
+    shift 1
```

Comment:
Maybe just pass `$1` here rather than `$@`? And shift earlier? So something like:

```suggestion
    ORIGINAL_COMMAND=$1
    shift 1
    set_main_variables "$ORIGINAL_COMMAND" || return 1
```

Then the shifting is in the same place as where `$1` gets meaning (not where it is interpreted, that is `set_main_variables`, but that's ok).

Reply:
Done.